src/framework/ui/button/button.component.tsx

```diff
@@ -93,15 +91,10 @@ export class Button extends React.Component<Props> {
   };
 
   private createImageElement = (style: StyleType): React.ReactElement<ImageProps> => {
-    const { icon: image } = this.props;
-
-    return (
-      <Image
-        key={0}
-        style={[style, strictStyles.icon]}
-        source={image}
-      />
-    );
+    const { icon } = this.props;
+    return icon && React.cloneElement(icon(style.width, style.height, style.color), {
```

Comment:
* prefer using control flow operators like `if`
* assign icon value to variable like `iconElement`

Reply:
Done.